storage/app/src/main/java/com/google/firebase/referencecode/storage/KotlinStorageActivity.kt

```diff
@@ -0,0 +1,405 @@
+        spaceRef.bucket
+        // [END reference_properties]
+
+        // ## Full Example
+
+        // [START reference_full_example]
+        // Points to the root reference
+        storageRef = storage.reference
+
+        // Points to "images"
+        imagesRef = storageRef.child("images")
+
+        // Points to "images/space.jpg"
+        // Note that you can use variables to create child values
+        val fileName = "space.jpg"
+        spaceRef = imagesRef.child(fileName)
+
+        // File path is "images/space.jpg"
+        val path = spaceRef.path
+
+        // File name is "space.jpg"
+        val name = spaceRef.name
+
+        // Points to "images"
+        imagesRef = spaceRef.parent
+        // [END reference_full_example]
+    }
+
+    override fun includesForUploadFiles() {
+        val storage = FirebaseStorage.getInstance()
+
+        // [START upload_create_reference]
+        // Create a storage reference from our app
+        val storageRef = storage.reference
+
+        // Create a reference to "mountains.jpg"
+        val mountainsRef = storageRef.child("mountains.jpg")
+
+        // Create a reference to 'images/mountains.jpg'
+        val mountainImagesRef = storageRef.child("images/mountains.jpg")
+
+        // While the file names are the same, the references point to different files
+        mountainsRef.name == mountainImagesRef.name    // true
+        mountainsRef.path == mountainImagesRef.path    // false
+        // [END upload_create_reference]
+
+
+        val imageView = findViewById<ImageView>(android.R.id.text1)
+
+        // [START upload_memory]
+        // Get the data from an ImageView as bytes
+        imageView.isDrawingCacheEnabled = true
+        imageView.buildDrawingCache()
+        val bitmap = (imageView.drawable as BitmapDrawable).bitmap
+        val baos = ByteArrayOutputStream()
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, baos)
+        val data = baos.toByteArray()
+
+        var uploadTask = mountainsRef.putBytes(data)
+        uploadTask.addOnFailureListener { exception ->
+            // Handle unsuccessful uploads
+        }.addOnSuccessListener { taskSnapshot ->
+            // taskSnapshot.metadata contains file metadata such as size, content-type, etc.
+            // ...
+        }
+        // [END upload_memory]
+
+        // [START upload_stream]
+        val stream = FileInputStream(File("path/to/images/rivers.jpg"))
+
+        uploadTask = mountainsRef.putStream(stream)
+        uploadTask.addOnFailureListener { exception ->
+            // Handle unsuccessful uploads
+        }.addOnSuccessListener { taskSnapshot ->
+            // taskSnapshot.metadata contains file metadata such as size, content-type, etc.
+            // ...
+        }
+        // [END upload_stream]
+
+        // [START upload_file]
+        var file = Uri.fromFile(File("path/to/images/rivers.jpg"))
+        val riversRef = storageRef.child("images/"+file.lastPathSegment)
```

Comment:
nit: spaces around the `+` sign

Reply:
I'm actually considering using the appropriate [String template](https://kotlinlang.org/docs/reference/basic-syntax.html#using-string-templates), so it would become `storageRef.child("images/${file.lastPathSegment}")`

Reply:
That sounds even better